pulsar-client-cpp/lib/MessageIdImpl.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
```

Comment:
"#pragma once" is compiler specific and potentially not portable. "#ifndef/#define/#endif" is more portable and will work for all preprocessors.

Reply:
Correct, however, we should decide whether we use it based on platforms we support. Otherwise libraries that we can use would be limited too much.

FWIW, Traffic Server project is going to use "#pragma once" (and C++11), we haven't changed them yet though.
https://lists.apache.org/thread.html/080f5eaf7b6bf26788affa8bddbfaf99abeb1673e739fcd79ed2bd1e@%3Cdev.trafficserver.apache.org%3E

Reply:
Supported compilers:
https://en.wikipedia.org/wiki/Pragma_once

Also, from our docs:
> We welcome contributions from the open source community, kindly make sure your changes are backward compatible with gcc-4.4.7 and Boost 1.41.

Reply:
> We welcome contributions from the open source community, kindly make sure your changes are backward compatible with gcc-4.4.7 and Boost 1.41.

The murmur hash function changes already break the Boost 1.41 part of the statement 

But yes in general looks like pragma once is highly used and has been there for a long time. So I have no objection to it being used.